### PR TITLE
`bool_to_int_with_if` inverse case patch

### PR DESCRIPTION
**Pull request by Xaeroxe**
Enhances `bool_to_int_with_if` such that it can also catch an inverse bool int conversion scenario, and makes the right suggestion for converting to int with a prefixed negation operator.

changelog: [`bool_to_int_with_if`]: Now correctly detects the inverse case, `if bool { 0 } else { 1 }`

### CODE REVIEW

File: clippy_lints/src/bool_to_int_with_if.rs [hunk 1]
```diff
@@ -55,27 +55,43 @@ fn check_if_else<'tcx>(ctx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx
...
+            true
+        } else {
+            // Expression isn't boolean, exit
+            return;
+        };
         let mut applicability = Applicability::MachineApplicable;
         let snippet = snippet_block_with_applicability(ctx, check.span, "..", None, &mut applicability);
+
+        let invert = if inverted { "!" } else { "" };
+        let need_parens = should_have_parentheses(check);
```
Comment (kraktus) [inline clippy_lints/src/bool_to_int_with_if.rs:75]: Clippy has an helper struct that will already handle parenthesis for you, including when negating the expression: [clippy_utils](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/index.html)::[sugg](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/sugg/index.html)::[Sugg](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/sugg/enum.Sugg.html#)

Comment (Xaeroxe) [reply clippy_lints/src/bool_to_int_with_if.rs]: That fixed a bug in the lint, and in turn fixed a bug in `clippy_utils::sugg::Sugg`, so thanks!

[bug in `Sugg`](https://github.com/rust-lang/rust-clippy/pull/9476/commits/dd97c1ed205f47d13dc5bc482a73c0c76383d159)

**Review by xFrednet — APPROVED** [review]

**Comment by xFrednet** [comment]
This looks good to me, thank you for this enhancement! I hope you had fun getting into Clippy! Furthermore, thanks to @kraktus for the review and suggestion. :upside_down_face: 

@bors r+

**Comment by Xaeroxe** [comment]
> I hope you had fun getting into Clippy!

I did! Thank you

**Pull request merged**